packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -83,16 +83,21 @@ impl<'id> MemoryId<'id> {
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
     pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
+        // We're ignoring glob errors as they shouldn't stop the whole cleanup process.
+        //
         // TODO: macOS does not store the shared memory FDs at `/dev/shm/`. Maybe it's not storing
         //   FDs at all. Find out if they are stored somewhere and remove them instead, otherwise we
         //   have to figure out a way to remove them without relying on the file-system.
-        let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)));
-
-        shm_files.into_iter().flatten().flatten().for_each(|path| {
-            if let Err(err) = std::fs::remove_file(&path) {
-                tracing::warn!("Could not clean up {path:?}: {err}");
-            }
-        });
+        if let Ok(shm_files) = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id))) {
+            shm_files
+                .into_iter()
+                .filter_map(Result::ok)
```

Comment:
I'm also not comfortable with us **silently** ignoring errors, this isn't good practice. Maybe something like:
```rust
shm_files
    .into_iter()
    .filter_map(|glob_res| match glob_res {
        Ok(path) => Some(path),
        Err(err) => {
            tracing::warn!(
                "Glob Error while trying to clean-up shared memory files: {E}"
            );
            None
        }
    })
```

Reply:
Applied in https://github.com/hashintel/hash/pull/494/commits/1d06d0fb07540c2514eb04225c48d2e516a37f73